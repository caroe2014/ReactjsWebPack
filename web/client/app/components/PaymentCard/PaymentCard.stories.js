// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import { storiesOf } from '@storybook/react';
import PaymentCard from '.';

storiesOf('PaymentCard', module)
  .add('Visa With Image', () => {
    const paymentOptions = {
      image: 'https://cdn.freebiesupply.com/logos/large/2x/visa-logo-png-transparent.png',
      id: '101',
      name: 'CREDIT/ DEBIT CARD',
      value: 'cardpay',
      valid: true
    };
    return (
      <PaymentCard keyProps={paymentOptions} />
    );
  }
  )
  .add('Apple Pay', () => {
    const paymentOptions = {
      image: 'https://image.flaticon.com/icons/svg/825/825455.svg',
      id: '102',
      name: 'APPLE PAY',
      value: 'applepay',
      valid: true
    };
    return (
      <PaymentCard keyProps={paymentOptions} />
    );
  }
  )
  .add('Android Pay', () => {
    const paymentOptions = {
      image: ' http://en.prothomalo.com/contents/cache/images/350x0x1/uploads/media/2018/02/22/fb830ec3a1e1390306c609af7ae4a6cd-Google-Pay.png', // eslint-disable-line max-len
      id: '103',
      name: 'ANDROID PAY',
      value: 'androidpay',
      valid: false
    };
    return (
      <PaymentCard keyProps={paymentOptions} />
    );
  }
  )
  .add('Without Image', () => {
    const paymentOptions = {
      id: '101',
      name: 'CREDIT/ DEBIT CARD',
      value: 'cardpay',
      valid: true
    };
    return (
      <PaymentCard keyProps={paymentOptions} />
    );
  }
  );
