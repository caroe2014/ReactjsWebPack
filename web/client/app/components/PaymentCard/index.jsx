// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.
/* eslint-disable no-mixed-operators */
/* eslint-disable max-len */
import React, { Component } from 'react';
import styled, { ThemeProvider } from 'styled-components';
import theme from 'web/client/theme';
import PropTypes from 'prop-types';
import { Flex, Text } from 'rebass';
import config from 'app.config';
import UrlImageLoader from 'web/client/app/components/UrlImageLoader';
import { Trans } from 'react-i18next';
import { currencyLocaleFormat } from 'web/client/app/utils/NumberUtils';
import { LoadingComponent } from 'web/client/app/components/Loader/ReusableLoader';
import i18n from 'web/client/i18n';

const Container = styled(Flex)`
  align-items: left;
  flex-wrap: no-wrap;
  flex-direction: column;
  min-height: 50px;
  max-height: 100px;
  justify-content: center;
  margin: 0.5em 1em 0.5em 1em;
  ${props => props.theme.mediaBreakpoints.mobile`
    margin: 0em auto 1em auto;
  `};
  ${props => props.theme.mediaBreakpoints.tablet`
    margin: 0em auto 1em auto;
  `};
  width: 100%;
  max-width: 280px;
  border: 2px solid ${props => props.theme.colors.buttonControlColor};
  border-radius: 6px;
  padding: ${props => {
    if (props.amountcharged || props.showLoading) return '4px 0px 0px';
    return '4px 0px 4px';
  }};
  opacity: ${props => {
    if (props.validpayment) return '1';
    return '0.4';
  }};
  }
  color: ${props => props.theme.colors.buttonControlColor};
  &:hover {
    cursor: ${props => props.validpayment && !props.showLoading ? 'pointer' : 'initial'};
    .account-container{
      color: ${props => props.theme.colors.buttonControlColor};
    }
    
${props => {
    if (props.validpayment) {
      return `box-shadow: 2px 3px 10px 0px rgba(0, 0, 0, .30);
        -webkit-box-shadow: 2px 3px 10px 0px rgba(0, 0, 0, .30);
        -moz-box-shadow: 2px 3px 10px 0px rgba(0, 0, 0, .30);`;
    }
  }};
`;

const DetailContainer = styled(Flex)`
  justify-content: center;
  align-items: flex-start;
  width: 100%;
  flex: 0.7;
`;

const BoxContainer = styled(Flex)`
  justify-content: center;
  align-items: center;
  flex: 0.3;
`;

const TextContainer = styled(Text)`
  font-size: 14px;
  font-weight: bold;
  text-align: left;
  ::first-letter {
    text-transform: uppercase;
  }
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const InstructionText = styled(Text)`
  font-size: 14px;
  font-weight: 500;
  text-align: left;
  margin-top: 4px;
  overflow: hidden;
  white-space: nowrap;
  text-overflow: ellipsis;
`;

const GATextContainer = styled(Text)`
  font-size: 14px;
  font-weight: 500;
  text-align: left;
  margin-top: 4px;
`;

const AccountContainer = styled(Flex)`
  flex-direction: row;
  width: 100%;
  justify-content: center;
  align-items: center;
  margin-bottom: ${props => {
    if (props.amountcharged || props.showLoading) return '4px';
    return '0px';
  }};
  height: 40px;
`;
const AmountContainer = styled(Flex)`
  width: 100%;
  justify-content: center;
  align-items: center;
  background: ${props => props.theme.colors.buttonControlColor};
  color: ${props => props.theme.colors.buttonTextColor};
  height: 30px;
  &:focus {
    outline: none;
    box-shadow: none;
  }
`;

const LoadingContainer = styled(Flex)`
  width: 100%;
  justify-content: center;
  align-items: center;
  background: ${props => props.theme.colors.buttonControlColor};
  color: ${props => props.theme.colors.buttonTextColor};
  height: 30px;
`;

const AtriumContainer = styled(Flex)`
  width: 100%;
  justify-content: space-between;
  align-items: center;
  padding-right: 5px;
`;

const RightText = styled(Text)`
  color: ${props => props.theme.colors.primaryTextColor};
  max-width: 40%;
  line-height: 1em;
  white-space: normal;
  overflow-wrap: break-word;
`;

const cardLogoStyle = {
  width: '40px',
  height: '40px',
  backgroundSize: 'cover',
  backgroundRepeat: 'no-repeat',
  maxHeight: '40px'
};

const fontStyle = {
  fontSize: '40px'
};

class PaymentCard extends Component {
  constructor (props) {
    super(props);
    this.state = {
      hover: false,
      newImage: ''
    };
    this.handleOption = this.handleOption.bind(this);
    this.filterPaymentTypes = this.filterPaymentTypes.bind(this);
    this.handleKeyDown = this.handleKeyDown.bind(this);
    this.setAriaLabel = this.setAriaLabel.bind(this);
  }

  handleOption (selectedTile, paymentObj) {
    !this.props.isPaymentProcess && this.props.onSelectOptionList(selectedTile, paymentObj);
  }

  handleKeyDown (event, selectedTile, paymentObj) {
    // TODO: make this usable by all clickable elements,
    //  pass in caller's onclickhandler (13 is keycode for enter key, 32 is spacebar)
    if (event.which === 13 || event.which === 32) {
      this.handleOption(selectedTile, paymentObj);
    }
  }

  paymentFont (type) {
    if (type === 'rGuestIframe' || type === 'savedCard' || type === 'ccCard') {
      return 'credit_card';
    } else if (type === 'genericAuthorization') {
      return 'account_box';
    } else if (type === 'roomCharge') {
      return 'room_service';
    } else if (type === 'memberCharge') {
      return 'card_membership';
    } else if (type.includes('loyalty')) {
      return 'LoyaltyAccrual';
    }
  }

  returnLoyaltyDetails (loyaltyAccount) {
    const accountNumberLast4 = loyaltyAccount.primaryAccountId && loyaltyAccount.primaryAccountId.substring(loyaltyAccount.primaryAccountId.length - 4);
    // TODO: pull Casino loyalty name from configuration
    const paymentOptionName = `Casino Loyalty ${accountNumberLast4 && '...' + accountNumberLast4}`;

    return (
      <React.Fragment>
        <TextContainer
          className='text-container'
          data-color={(this.state.hover && loyaltyAccount.valid) ||
            (this.props.selectedOptionId === loyaltyAccount.type && loyaltyAccount.valid)}
        >
          {paymentOptionName}
        </TextContainer>
        <GATextContainer
          className='text-container'
          data-color={(this.state.hover && loyaltyAccount.valid) ||
            (this.props.selectedOptionId === loyaltyAccount.type && loyaltyAccount.valid)}
        >
          {loyaltyAccount.displayLabel}
        </GATextContainer>
      </React.Fragment>
    );
  }
  setAriaLabel (keyProps) {
    if (keyProps.type === 'loyaltyHostCompVoucher' || keyProps.type === 'loyaltyPoints' || keyProps.type === 'loyaltyVoucher') {
      const accountNumberLast4 = (keyProps.type === 'loyaltyHostCompVoucher' || keyProps.type === 'loyaltyPoints' || keyProps.type === 'loyaltyVoucher') ? keyProps.primaryAccountId && keyProps.primaryAccountId.substring(keyProps.primaryAccountId.length - 4) : '';
      // TODO: pull Casino loyalty name from configuration
      const paymentOptionName = `Casino Loyalty ${accountNumberLast4 && '...' + accountNumberLast4}`;
      return `${paymentOptionName} ${keyProps.displayLabel}`;
    } else if (keyProps.type === 'gaAccount') {
      return `${i18n.t('GA_ACCOUNT_LABEL')}  ${keyProps.primaryAccountId !== undefined ? keyProps.primaryAccountId : ''}${keyProps.displayLabel}`;
    } else {
      return keyProps.displayLabel;
    }
  }

  returnAtriumDetails (atriumAccount) {

    return (
      <AtriumContainer>
        <Flex flexDirection='column' style={{overflow: 'hidden',
          whiteSpace: 'nowrap',
          textOverflow: 'ellipsis'}}>
          <TextContainer
            style={{overflow: 'hidden',
              whiteSpace: 'nowrap',
              textOverflow: 'ellipsis'}}
            className='text-container'
          >
            { atriumAccount.displayLabel }
          </TextContainer>
          {atriumAccount.instructionText && <InstructionText
            className='instruction-text'
          >
            {atriumAccount.instructionText}
          </InstructionText>
          }
        </Flex>
        {!atriumAccount.isAutoDetect &&
          atriumAccount.authResponse
          ? <RightText>
            {atriumAccount.limitOnAccount ? atriumAccount.authResponse.paymentData.paymentResponse.transactionData.atriumPaymentResponse.amount.remaining
              : currencyLocaleFormat(atriumAccount.authResponse.paymentData.paymentResponse.transactionData.atriumPaymentResponse.amount.remaining)}
          </RightText>
          : (atriumAccount.amount && atriumAccount.amount.remaining) && <RightText>
            {atriumAccount.limitOnAccount ? atriumAccount.amount.remaining : currencyLocaleFormat(atriumAccount.amount.remaining)}
          </RightText>}
      </AtriumContainer>
    );
  }

  filterPaymentTypes () {
    const { keyProps } = this.props;

    return keyProps.type !== 'gaAccount' && keyProps.type !== 'loyaltyHostCompVoucher' &&
      keyProps.type !== 'loyaltyPoints' && keyProps.type !== 'loyaltyVoucher' && keyProps.type !== 'atrium';
  }

  render () {
    const { keyProps, currencyDetails } = this.props;
    const { classContext } = this.props;
    const image = keyProps.type === 'genericAuthorization' || keyProps.type === 'gaAccount' ? keyProps.tenderTypeImage : keyProps.image;
    const amountToBeCharged = keyProps.amountToBeCharged && currencyLocaleFormat(keyProps.amountToBeCharged, currencyDetails);
    const ariaLabel = this.setAriaLabel(keyProps);

    return (
      <ThemeProvider className='card' theme={theme}>
        <Container
          className={`click-cont container ${classContext}`}
          alignItems='center'
          flexWrap='nowrap'
          amountcharged={keyProps.amountToBeCharged}
          validpayment={keyProps.valid ? 1 : 0}
          selectedoption={this.props.selectedOptionId}
          currentoption={keyProps.type}
          onClick={() => { keyProps.valid && this.handleOption(keyProps.type, keyProps); }}
          onKeyDown={(e) => { keyProps.valid && this.handleKeyDown(e, keyProps.type, keyProps); }}
          onMouseEnter={() => this.setState({ hover: true })}
          onMouseLeave={() => this.setState({ hover: false })}
          showLoading={this.props.showLoading}
          role='button'
          aria-label={ariaLabel}
          tabIndex={0}
        >
          <AccountContainer className='account-container' amountcharged={keyProps.amountToBeCharged}>

            <BoxContainer className='box-container' width={[1]} >
              {image ? <UrlImageLoader
                className='url-container'
                src={config.getPOSImageURL(this.props.contextId, image, this.props.appconfig.tenantID)}
                imageStyle={cardLogoStyle}
                alt='' />
                : <i
                  className={`fa agilysys-icon-${this.paymentFont(keyProps.type)}`}
                  style={keyProps.type.includes('loyalty') ? { fontSize: '32px' } : fontStyle}
                  alt=''
                />}
            </BoxContainer>
            <DetailContainer
              className={`detail-container-${keyProps.type}`}
              flexDirection='column'
              alignItems='flex-start'
              imageflag={image}
              width={[1]}
            >
              {keyProps.type === 'gaAccount' &&
                <React.Fragment>
                  <TextContainer
                    className='text-container'
                    data-color={(this.state.hover && keyProps.valid) ||
                      (this.props.selectedOptionId === keyProps.type && keyProps.valid)}
                  >
                    <Trans i18nKey='GA_ACCOUNT_LABEL' />{keyProps.primaryAccountId}
                  </TextContainer>
                  <GATextContainer
                    className='text-container'
                    data-color={(this.state.hover && keyProps.valid) ||
                      (this.props.selectedOptionId === keyProps.type && keyProps.valid)}
                  >
                    {keyProps.displayLabel}
                  </GATextContainer>
                </React.Fragment>
              }
              {(keyProps.type === 'loyaltyHostCompVoucher' || keyProps.type === 'loyaltyPoints' || keyProps.type === 'loyaltyVoucher') &&
                this.returnLoyaltyDetails(keyProps)
              }
              { (keyProps.type === 'atrium') &&
                this.returnAtriumDetails(keyProps)
              }
              {
                this.filterPaymentTypes() &&
                <TextContainer
                  className='text-container'
                  data-color={(this.state.hover && keyProps.valid) ||
                    (this.props.selectedOptionId === keyProps.type && keyProps.valid)}
                >
                  {keyProps.displayLabel}
                </TextContainer>
              }

            </DetailContainer>
          </AccountContainer>
          {
            keyProps.amountToBeCharged &&
            <AmountContainer className='amount-charged'
              tabIndex={0}
              aria-label={amountToBeCharged}>
              {amountToBeCharged}
            </AmountContainer>
          }
          {
            this.props.showLoading &&
            <LoadingContainer>
              <LoadingComponent
                className='loading-cont'
                color='white'
                containerHeight='100%'
                containerWidth='100%'
                aria-label={i18n.t('PROCESSING_TEXT')}
                height='25px'
                width='25px'
                borderSize={2}
                style={{ justifyContent: 'center', marginBottom: '6px' }}
              />
            </LoadingContainer>
          }
        </Container>
      </ThemeProvider>
    );
  }
}

PaymentCard.propTypes = {
  keyProps: PropTypes.shape({
    image: PropTypes.string,
    displayLabel: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    valid: PropTypes.bool.isRequired
  }).isRequired
};

export default PaymentCard;
