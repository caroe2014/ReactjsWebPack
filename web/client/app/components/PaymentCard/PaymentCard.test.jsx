// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React from 'react';
import {
  shallow,
  mount,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import {
  ThemeProvider
} from 'styled-components';
import theme from 'web/client/theme';
import { t } from 'i18next';

import PaymentCard from '.';

configure({
  adapter: new Adapter()
});

jest.mock('web/client/i18n', () => ({
  t: (key) => {
    return key;
  }
}));

jest.mock('react-i18next', () => ({
  Trans: 'div'
}));

jest.mock('../UrlImageLoader', () => ()=> <div id="mockUrlImageLoaderm">
   mockUrlImageLoader
</div>);

const MOCK_APP_CONFIG = {
  contextID: '1',
  tenantID: '1'
};

describe('PaymentCard', () => {
  let mockProps = {
    image: 'https://cdn.freebiesupply.com/logos/large/2x/visa-logo-png-transparent.png',
    id: '101',
    displayLabel: 'CREDIT/ DEBIT CARD',
    name: 'CREDIT/ DEBIT CARD',
    value: 'cardpay',
    type: 'cardpay',
    valid: true
  }
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PaymentCard keyProps={mockProps} appconfig={MOCK_APP_CONFIG} />);
  });

  it('does a snapshot test', () => {
    const snapWrapper = mount(<ThemeProvider theme={theme}><PaymentCard keyProps={mockProps} appconfig={MOCK_APP_CONFIG} /></ThemeProvider>);
    expect(toJson(snapWrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(wrapper.exists()).toBe(true);
  });

  it('should disable if invalid', () => {
    let invalidMock = {
      image: 'https://cdn.freebiesupply.com/logos/large/2x/visa-logo-png-transparent.png',
      id: '101',
      displayLabel: 'CREDIT/ DEBIT CARD',
      name: 'CREDIT/ DEBIT CARD',
      value: 'cardpay',
      type: 'cardpay',
      valid: false
    }
    const invalidWrapper = shallow(<ThemeProvider theme={theme}><PaymentCard keyProps={invalidMock} appconfig={MOCK_APP_CONFIG} /></ThemeProvider>);
    expect(invalidWrapper.exists()).toBe(true);
  });

  it('should show if type is rGuestIframe', () => {
    let iFrameMock = {
      image: 'https://cdn.freebiesupply.com/logos/large/2x/visa-logo-png-transparent.png',
      id: '101',
      displayLabel: 'CREDIT/ DEBIT CARD',
      name: 'CREDIT/ DEBIT CARD',
      value: 'cardpay',
      type: 'rGuestIframe',
      valid: true
    }
    const iFrameWrapper = mount(<ThemeProvider theme={theme}><PaymentCard keyProps={iFrameMock} selectedOptionId={'rGuestIframe'} appconfig={MOCK_APP_CONFIG} /></ThemeProvider>);
    expect(iFrameWrapper.exists()).toBe(true);
  });

  it('should show simulate mouseover', () => {
    let iFrameMock = {
      id: '101',
      displayLabel: 'CREDIT/ DEBIT CARD',
      name: 'CREDIT/ DEBIT CARD',
      value: 'cardpay',
      type: 'rGuestIframe',
      valid: false
    }
    const iFrameWrapper = shallow(<PaymentCard keyProps={iFrameMock} selectedOptionId={'rGuestIframe'} appconfig={MOCK_APP_CONFIG} />);
    const inputMobile = iFrameWrapper.find('.container');
    inputMobile.simulate('mouseover');
  });

  // it('checks whether image prop is recevied', () => {
  //   expect(wrapper.find('.detail-container').props().imageflag).toBe(mockProps.image);
  // });

  describe('when Image prop is undefined', () => {
    let wrapper;

    beforeEach(() => {
      mockProps.image = '';
      wrapper = shallow(<PaymentCard keyProps={mockProps} appconfig={MOCK_APP_CONFIG}/>);
    });

    it('checks whether image prop is not recevied', () => {
      expect(wrapper.find('.image').length).toBe(0);
    });
  });

  describe('when mouse is over the component', () => {
    it('should highlight based on hover state', () => {
      const wrapper = shallow(<PaymentCard keyProps={mockProps} appconfig={MOCK_APP_CONFIG}/>);
      expect(wrapper.state().hover).toBe(false);
      const input = wrapper.find('.container');
      input.simulate('mouseLeave');
      expect(wrapper.state().hover).toBe(false);
      input.simulate('mouseEnter');
      expect(wrapper.state().hover).toBe(true);
    });
  });

  it('should call handleDelete when DeleteButton is clicked', () => {
    let mockFn = jest.fn();
    let wrapper = shallow(<PaymentCard keyProps={mockProps} onSelectOptionList={mockFn} appconfig={MOCK_APP_CONFIG}/>);
    const spy = jest.spyOn(wrapper.instance(), 'handleOption');
    wrapper.instance().forceUpdate();
    wrapper.find('.click-cont').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should render Loader when loading state is true', () => {
    let wrapper = mount(<ThemeProvider theme={theme}><PaymentCard keyProps={mockProps} appconfig={MOCK_APP_CONFIG} /></ThemeProvider>);
    wrapper.setState({ keyProps: {type: 'applePay'} });
    wrapper.instance().forceUpdate();
    expect(wrapper.exists()).toBe(true);
  });

});
