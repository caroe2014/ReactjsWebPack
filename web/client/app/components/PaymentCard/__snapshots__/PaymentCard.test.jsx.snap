// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentCard does a snapshot test 1`] = `
<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <PaymentCard
    appconfig={
      Object {
        "contextID": "1",
        "tenantID": "1",
      }
    }
    keyProps={
      Object {
        "displayLabel": "CREDIT/ DEBIT CARD",
        "id": "101",
        "image": "https://cdn.freebiesupply.com/logos/large/2x/visa-logo-png-transparent.png",
        "name": "CREDIT/ DEBIT CARD",
        "type": "cardpay",
        "valid": true,
        "value": "cardpay",
      }
    }
  >
    <ThemeProvider
      className="card"
      theme={
        Object {
          "colors": Object {
            "addOnTab": "#347CFF",
            "addToCart": "#347CFF",
            "addonBg": "#EEEEEE",
            "backgroundGrey": "#E6E6E6",
            "black": "#000000",
            "close": "red",
            "contextBackground": "#f0f0f3",
            "darkGrey": "#464646",
            "disableBackground": "#E4E4E4",
            "disableButton": "#D1D1D1",
            "error": "#cc0000",
            "footer": "#48a942",
            "glutenFree": "#8d7356",
            "healthy": "#359919",
            "hint": "#a6a6a6",
            "inputBottomBorder": "lightgray",
            "itemBorder": "lightgrey",
            "itemTile": "#3C474C0",
            "light": "white",
            "loader": "#ababab",
            "loginBackground": "#303030",
            "loginBoxColor": "#434149",
            "loginField": "#AAA7AC",
            "loginFormBackground": "#484848",
            "navigationMenu": "#48a942",
            "navigationMenuHover": "cornflowerblue",
            "new": "#ff6000",
            "open": "#48a942",
            "paymentSuccessTitle": "#505050",
            "placeHolder": "#909090",
            "primary": "#347CFF",
            "primaryButton": "#347CFF",
            "primaryText": "#383838",
            "secondaryText": "#a6a6a6",
            "spicy": "#ed1c24",
            "text": "black",
            "textGrey": "#505050",
            "tile": "#505050",
            "validationError": "#cc0000",
            "vegan": "#0d4a22",
            "vegetarian": "#0d4a22",
          },
          "fontFamily": "Roboto",
          "fontSize": Object {
            "lg": "1.8rem",
            "md": "1.3rem",
            "nm": "1rem",
            "sm": "0.75rem",
            "xl": "2.4rem",
          },
          "fontSizeFromProps": [Function],
          "mediaBreakpoints": Object {
            "desktop": [Function],
            "mobile": [Function],
            "tablet": [Function],
          },
          "nav": Object {
            "height": "70px",
          },
          "receiptFontFamily": "Roboto",
          "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
          "spacing": Object {
            "lg": "4em",
            "md": "2em",
            "nm": "1em",
            "none": "0em",
            "sm": "0.5em",
            "xl": "8em",
          },
        }
      }
    >
      <Styled(Flex)
        alignItems="center"
        aria-label="CREDIT/ DEBIT CARD"
        className="click-cont container undefined"
        currentoption="cardpay"
        flexWrap="nowrap"
        onClick={[Function]}
        onKeyDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        role="button"
        tabIndex={0}
        validpayment={1}
      >
        <Flex
          alignItems="center"
          aria-label="CREDIT/ DEBIT CARD"
          className="click-cont container undefined sc-uJMKN bOSvrq"
          currentoption="cardpay"
          flexWrap="nowrap"
          onClick={[Function]}
          onKeyDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="button"
          tabIndex={0}
          validpayment={1}
        >
          <Box
            alignItems="center"
            aria-label="CREDIT/ DEBIT CARD"
            className="click-cont container undefined sc-uJMKN bOSvrq sc-bwzfXH cjHxAH"
            currentoption="cardpay"
            flexWrap="nowrap"
            onClick={[Function]}
            onKeyDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            role="button"
            tabIndex={0}
            validpayment={1}
          >
            <div
              aria-label="CREDIT/ DEBIT CARD"
              className="click-cont container undefined sc-uJMKN bOSvrq sc-bwzfXH cjHxAH sc-bdVaJa iHZvIS"
              onClick={[Function]}
              onKeyDown={[Function]}
              onMouseEnter={[Function]}
              onMouseLeave={[Function]}
              role="button"
              tabIndex={0}
            >
              <Styled(Flex)
                className="account-container"
              >
                <Flex
                  className="account-container sc-hEsumM dYNwUI"
                >
                  <Box
                    className="account-container sc-hEsumM dYNwUI sc-bwzfXH hKiLMS"
                  >
                    <div
                      className="account-container sc-hEsumM dYNwUI sc-bwzfXH hKiLMS sc-bdVaJa iHZvIS"
                    >
                      <Styled(Flex)
                        className="box-container"
                        width={
                          Array [
                            1,
                          ]
                        }
                      >
                        <Flex
                          className="box-container sc-gGBfsJ cJuXXz"
                          width={
                            Array [
                              1,
                            ]
                          }
                        >
                          <Box
                            className="box-container sc-gGBfsJ cJuXXz sc-bwzfXH hKiLMS"
                            width={
                              Array [
                                1,
                              ]
                            }
                          >
                            <div
                              className="box-container sc-gGBfsJ cJuXXz sc-bwzfXH hKiLMS sc-bdVaJa gRrvFh"
                              width={
                                Array [
                                  1,
                                ]
                              }
                            >
                              <Component
                                alt=""
                                className="url-container"
                                imageStyle={
                                  Object {
                                    "backgroundRepeat": "no-repeat",
                                    "backgroundSize": "cover",
                                    "height": "40px",
                                    "maxHeight": "40px",
                                    "width": "40px",
                                  }
                                }
                                src="application/api/image/1/undefined/https://cdn.freebiesupply.com/logos/large/2x/visa-logo-png-transparent.png"
                              >
                                <div
                                  id="mockUrlImageLoaderm"
                                >
                                  mockUrlImageLoader
                                </div>
                              </Component>
                            </div>
                          </Box>
                        </Flex>
                      </Styled(Flex)>
                      <Styled(Flex)
                        alignItems="flex-start"
                        className="detail-container-cardpay"
                        flexDirection="column"
                        imageflag="https://cdn.freebiesupply.com/logos/large/2x/visa-logo-png-transparent.png"
                        width={
                          Array [
                            1,
                          ]
                        }
                      >
                        <Flex
                          alignItems="flex-start"
                          className="detail-container-cardpay sc-bbmXgH ljmMSZ"
                          flexDirection="column"
                          imageflag="https://cdn.freebiesupply.com/logos/large/2x/visa-logo-png-transparent.png"
                          width={
                            Array [
                              1,
                            ]
                          }
                        >
                          <Box
                            alignItems="flex-start"
                            className="detail-container-cardpay sc-bbmXgH ljmMSZ sc-bwzfXH iaREIe"
                            flexDirection="column"
                            imageflag="https://cdn.freebiesupply.com/logos/large/2x/visa-logo-png-transparent.png"
                            width={
                              Array [
                                1,
                              ]
                            }
                          >
                            <div
                              className="detail-container-cardpay sc-bbmXgH ljmMSZ sc-bwzfXH iaREIe sc-bdVaJa gRrvFh"
                              width={
                                Array [
                                  1,
                                ]
                              }
                            >
                              <Styled(Text)
                                className="text-container"
                                data-color={false}
                              >
                                <Text
                                  className="text-container sc-jnlKLf iCKpPK"
                                  data-color={false}
                                  m={0}
                                >
                                  <ForwardRef
                                    className="text-container sc-jnlKLf iCKpPK sc-gqjmRU kKsieB"
                                    data-color={false}
                                    m={0}
                                  >
                                    <div
                                      className="text-container sc-jnlKLf iCKpPK sc-gqjmRU kKsieB"
                                      data-color={false}
                                    >
                                      CREDIT/ DEBIT CARD
                                    </div>
                                  </ForwardRef>
                                </Text>
                              </Styled(Text)>
                            </div>
                          </Box>
                        </Flex>
                      </Styled(Flex)>
                    </div>
                  </Box>
                </Flex>
              </Styled(Flex)>
            </div>
          </Box>
        </Flex>
      </Styled(Flex)>
    </ThemeProvider>
  </PaymentCard>
</ThemeProvider>
`;
